### DISCUSSION TITLE
Filter HTML reports to only show coverage results below a certain percentage

### DISCUSSION THREAD

**Discussion by bryansonofpatt** [danielpalme/ReportGenerator#662] [General]
Hello,
We use the Report Generator in our Azure DevOps pipeline. The output type is HtmlInline_AzurePipelines. We have a lot of classes we would like to ignore in the report whose coverage is above a certain percentage (90%). 
![image](https://github.com/danielpalme/ReportGenerator/assets/21263969/5bb5b8ca-617b-4f5f-8cf3-54e9bf61cd86)

I don't see a simple way to pass params to the Report Generator task to accomplish this. The only way I see is if someone wrote a custom plugin and put a % threshold in the CreateSummaryReport method of the IReportBuilder implementation. Am I correct in this assumption, or did I miss anything.

Thank you!
Bryan

**Comment by danielpalme** [comment]
Hi Bryan,

there is currently no way to filter classes based on their coverage.
I think a dynamic filter (e.g a slider) in the HTML report might be a good idea.

Only problem is, that this would not work within Azure DevOps, since JavaScript is blocked here.
It only would work if you download the report.

**Comment by danielpalme** [comment]
Release 5.3.0 now contains range sliders to dynamically filter classes based on coverage

![image](https://github.com/danielpalme/ReportGenerator/assets/859606/538a8b68-4eff-4fe4-b5de-536ab801c4c7)

**Comment by bryansonofpatt** [comment]
I ran v5.3 and still don't see the sliders. I also added 'publishCodeCoverageResults: true' param and removed 'PublishCodeCoverageResults' task. 
![image](https://github.com/danielpalme/ReportGenerator/assets/21263969/e968a4c8-36b9-40b3-ac04-d58f5d931cd6)
![image](https://github.com/danielpalme/ReportGenerator/assets/21263969/8e37015b-48f3-484d-8594-463faf574a57)
![image](https://github.com/danielpalme/ReportGenerator/assets/21263969/709ad38a-95d8-4f28-84be-32a45547cd97)

**Comment by danielpalme** [comment]
This only works outside the Azure DevOps "Code Coverage" tab.
Azure DevOps blocks all JavaScript and therefore the dynamic filtering.

Here's a sample report: https://reportgenerator.io/resources/reports/HtmlInline_AzurePipelines/index.html

**Comment by bryansonofpatt** [comment]
Good to know about the JavaScript in ADO. 
I downloaded all of the report artifacts that the tool generated and opened them in my browser. I did get some JavaScript - Collapse/Expand all, Grouping slider, column sorting, and Filter. I still do not see the percentage sliders though. Am I missing a param or config setting?

**ADO** 
![image](https://github.com/danielpalme/ReportGenerator/assets/21263969/4861d1d1-bc67-4665-9031-ff1fc22a9339)

**Downloaded HTML, rendered locally**
![image](https://github.com/danielpalme/ReportGenerator/assets/21263969/e8a72028-2447-4b4f-ac52-d7194704d885)

**Comment by danielpalme** [comment]
The screenshot indicate that you are not using the latest version 5.3.0.

Are you sure that the `PublishCodeCoverageResults` task has been removed?

If not, the report may get regenerated. See this wiki page for details:
https://github.com/danielpalme/ReportGenerator/wiki/Integration#attention

**Reply by bryansonofpatt** [reply]
oops! On my last pipeline run, I forgot to pick my POC branch.
We're good now!!! Thanks! 
![image](https://github.com/danielpalme/ReportGenerator/assets/21263969/3ea6d3d2-1b3f-42ca-88b3-72ffbdbf1311)
